packages/components/src/product-icon/README.md

```diff
@@ -0,0 +1,57 @@
+ProductIcon Component
+=============
+
+ProductIcon component is a React component to display plan or product icon in SVG format.
+
+Please note that this component uses asset imports for generating icon images URLs.
+Most build environments handle the asset imports gracefully out of the box (e.g. `create-react-app`, `calypso-build`).
+If you're using a custom Webpack environment, make sure that you have the [asset imports](https://webpack.js.org/guides/asset-management/#loading-images) set up correctly.
+
+## Usage
+
+```jsx
+import { ProductIcon } from '@automattic/components';
+
+export default function MyComponent() {
+    return (
+        <div className="my-component">
+            <ProductIcon slug="free" />
+            <ProductIcon slug="jetpack-professional" />
+            <ProductIcon slug="jetpack-backup-daily" />
+        </div>
+    );
+}
+
+```
+
+## Props
+
+### `slug`
+
+Plan or product slug. Can be one of the following:
+
+- `free`
+- `blogger`
+- `personal`
+- `premium`
+- `business`
+- `ecommerce`
+- `jetpack-free`
+- `jetpack-personal`
+- `jetpack-premium`
+- `jetpack-professional`
+- `jetpack-backup-daily`
+- `jetpack-backup-realtime`
+
+### `className`
+
+Additional class name to be added to the icon element.
+
+## How to Add New Icon
+
+In order to add a new icon to the component, do the following:
+* Add a new icon image to the `images` folder.
+* Import the image in the `paths.js` file. This way an image URL will be generated by a bundler for further reuse.
+* Add the image path to the default export block in the `paths.js` file.
```

Comment:
And another point you could consider adding here: `Optionally, update the docs/example.jsx file to include the new icon.`. That would be optional because one might want to not display that icon in the example sometimes.